overlord/ifacestate/ifacestate.go

```diff
@@ -84,39 +84,50 @@ func checkConnectConflicts(st *state.State, plugSnap, slotSnap string, auto bool
 		}
 
 		k := task.Kind()
-		if auto && k == "connect" {
-			var autoConnect bool
-			// the auto flag is set for connect tasks created as part of auto-connect
-			if err := task.Get("auto", &autoConnect); err != nil && err != state.ErrNoState {
+		if auto && (k == "connect" || k == "disconnect") {
+			plugRef, slotRef, err := getPlugAndSlotRefs(task)
+			if err != nil {
 				return err
 			}
-			// wait for connect task with "auto" flag if they affect our snap
-			if autoConnect {
+			if plugRef.Snap == plugSnap || slotRef.Snap == slotSnap {
+				return &state.Retry{After: connectRetryTimeout}
+			}
+		} else {
+			// FIXME: revisit this check for normal connects
```

Comment:
we are being more conservative blocking the ops now with this new code, the FIXME can go now I think